app/public/cantusdata/helpers/source_importer.py

```diff
@@ -0,0 +1,70 @@
+from urllib import request
+import json
+
+class SourceImporter:
+    """
+    Imports sources from CantusDB by way of its APIs.
+
+    :param str cantus_db_base_url: the domain for CantusDB.
+    """
+    def __init__(self, cantus_db_base_url):
+        self.cdb_base_url = cantus_db_base_url
+
+    def request_source_ids(self):
+        """
+        Requests the IDs of all sources available in CantusDB
+        
+        :return: a list of character source ID's
+        """
+        sources_url = self.cdb_base_url + "/json-sources/"
+        sources_response = request.urlopen(sources_url).read()
+        sources_json = json.loads(sources_response)
+        source_ids = sources_json.keys()
+        return source_ids
+
+    def request_source_data(self, source_id):
+        """
+        Requests data on an individual source in CantusDB using
+        the /json-node/ endpoint.
+        
+        :param str source_id: the ID of the source in CantusDB
+        :return: a parsed json object containing source data
+        """
+        source_url = self.cdb_base_url + "/json-node/" + source_id
+        source_response = request.urlopen(source_url).read()
+        source_json = json.loads(source_response)
+        return source_json
+
+    def source_json_extractor(source_json):
```

Comment:
this could be more verb-like? `extract_sources_from_json`? or even `extract_sources(json)`?

Reply:
also, we're missing an initial `self` parameter, aren't we?

Reply:
Function names improved